example-2/evm/src/contracts/lendingHub/HubWormholeUtilities.sol

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "../../libraries/external/BytesLib.sol";
+
+import "../../interfaces/IWormhole.sol";
+
+import "./HubGetters.sol";
+import "./HubSetters.sol";
+
+contract HubWormholeUtilities is HubGetters, HubSetters {
+    using BytesLib for bytes;
+
+    function transferTokens(address receiver, address assetAddress, uint256 amount, uint16 recipientChain)
+        internal
+        returns (uint64 sequence)
+    {
+        SafeERC20.safeApprove(IERC20(assetAddress), tokenBridgeAddress(), amount);
+        sequence = tokenBridge().transferTokens(
+            assetAddress, amount, recipientChain, bytes32(uint256(uint160(receiver))), 0, 0
+        );
+    }
+
+    function sendWormholeMessage(bytes memory payload) internal returns (uint64 sequence) {
+        sequence = wormhole().publishMessage(
+            0, // nonce
+            payload,
+            consistencyLevel()
+        );
+    }
+
+    function getTransferPayload(bytes memory encodedMessage) internal returns (bytes memory payload) {
+        (IWormhole.VM memory parsed,,) = wormhole().parseAndVerifyVM(encodedMessage);
+
+        verifySenderIsSpoke(
+            parsed.emitterChainId, address(uint160(uint256(parsed.payload.toBytes32(1 + 32 + 32 + 2 + 32 + 2))))
```

Comment:
Assigning a constant value for `1 + 32 + 32 + 2 + 32 + 2` might be useful here since you refer to this + 32 in code below.